src/jbi/bugzilla_objects.py

```diff
@@ -0,0 +1,247 @@
+    target: Optional[str]
+    routing_key: Optional[str]
+
+
+class BugzillaWebhookAttachment(BaseModel):
+    """Bugzilla Attachment Object"""
+
+    content_type: Optional[str]
+    creation_time: Optional[datetime.datetime]
+    description: Optional[str]
+    file_name: Optional[str]
+    flags: Optional[List]
+    id: int
+    is_obsolete: Optional[bool]
+    is_patch: Optional[bool]
+    is_private: Optional[bool]
+    last_change_time: Optional[datetime.datetime]
+
+
+class BugzillaWebhookComment(BaseModel):
+    """Bugzilla Comment Object"""
+
+    body: Optional[str]
+    id: Optional[int]
+    number: Optional[int]
+    is_private: Optional[bool]
+    creation_time: Optional[datetime.datetime]
+
+    def is_comment_description(self) -> bool:
+        """Used to determine if `self` is a description or comment."""
+        if self.number in [0]:  # Currently 0th comment is the description
+            return True
+        return False
+
+    def is_comment_generic(self) -> bool:
+        """All comments after comment-0 are generic"""
+        is_description = self.is_comment_description()
+        return not is_description
+
+    def is_private_comment(self) -> bool:
+        """Helper function to determine if this comment private--not accessible or open"""
+        if self.is_private is None:
+            self.is_private = False
+        return self.is_private
+
+
+class BugzillaBug(BaseModel):
+    """Bugzilla Bug Object"""
+
+    id: int
+    is_private: Optional[bool]
+    type: Optional[str]
+    product: Optional[str]
+    component: Optional[str]
+    whiteboard: Optional[str]
+    keywords: Optional[List]
+    flags: Optional[List]
+    status: Optional[str]
+    resolution: Optional[str]
+    see_also: Optional[List]
+    summary: Optional[str]
+    severity: Optional[str]
+    priority: Optional[str]
+    creator: Optional[str]
+    assigned_to: Optional[str]
+    comment: Optional[BugzillaWebhookComment]
+
+    def get_whiteboard_count(self):
+        """Get count of whiteboard labels"""
+        if self.whiteboard is not None:
+            return self.whiteboard.count("[")
+        return 0
+
+    def get_whiteboard_as_list(self):
+        """Convert string whiteboard into list, splitting on ']' and removing '['."""
+        if self.whiteboard is not None:
+            split_list = self.whiteboard.replace("[", "").split("]")
+            return [x.strip() for x in split_list if x not in ["", " "]]
+        return []
+
+    def get_whiteboard_with_brackets_as_list(self):
+        """Convert string whiteboard into list, splitting on ']' and removing '['; then re-adding."""
+        wb_list = self.get_whiteboard_as_list()
+        if wb_list is not None and len(wb_list) > 0:
+            return [f"[{element}]" for element in wb_list]
+        return []
+
+    def get_jbi_labels(self):
+        """bugzilla is a required label for SYNC project and is being used to prevent gh syncing"""
```

Comment:
gh syncing?

Reply:
Removed this message; it's a requirement by the Sync team--they have two syncing solutions github issues and jbi, they use this "bugzilla" tag to determine the origin of the issue for queries in jira